scripts/extensions/sams/src/apps/samsWorkspace.tsx

```diff
@@ -156,12 +156,22 @@ export class SamsWorkspaceComponent extends React.Component<IProps, IState> {
         this.onDownloadSingleAssetCompressedBinary = this.onDownloadSingleAssetCompressedBinary.bind(this);
         this.onMultiActionBar = this.onMultiActionBar.bind(this);
         this.onDeleteAsset = this.onDeleteAsset.bind(this);
+        this.onEditAsset = this.onEditAsset.bind(this);
     }
 
     onDeleteAsset(asset: IAssetItem): void {
         this.props.deleteAsset(asset);
     }
 
+    onEditAsset(asset: IAssetItem): void {
+        const updates: Dictionary<string, any> = {'lock_action': 'edit'};
```

Comment:
This might be better placed in a set of actions/reducers, with another action/reducer for each to update the Asset in the store.

This way after successfully locking/unlocking an item, the Asset in the redux store is also updated. This will then automatically update list views to pass the new lock status to item list/grid item component (as well as updating the `_etag` attribute)